src/H5FDsubfiling/H5FDsubfiling.c

```diff
@@ -2086,98 +1709,27 @@ H5FD__subfiling_read_vector(H5FD_t *_file, hid_t dxpl_id, uint32_t count, H5FD_m
     /* Set DXPL for operation */
     H5CX_set_dxpl(dxpl_id);
 
-    /* TODO: setup real support for vector I/O */
     if (file_ptr->fa.require_ioc) {
-
-        bool       extend_sizes = false;
```

Comment:
All this logic is now moved into the part of `H5FD__subfiling_io_helper` that handles the vector I/O requests.